portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -14,19 +14,92 @@
 
 package com.google.sps.servlets;
 
+import com.google.gson.Gson;
 import java.io.IOException;
+import java.util.ArrayList;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-/** Servlet that returns some example content. TODO: modify this file to handle comments data */
+
+/** 
+ * Servlet that returns some example content. TODO: modify this file to handle comments data 
+ */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+  private ArrayList<String> comments = new ArrayList<String>();
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+    // Convert the ArrayList to JSON
+    String json = convertToJsonUsingGson(comments);
+    
+    // Send the JSON as the response
+    response.setContentType("application/json;");
+    response.getWriter().println(json);
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+	// Get the input from the form
+    String name = getParameter(request, "name", "");
+	String email = getParameter(request, "email", "");
+    String text = getParameter(request, "comment", "");
+    
+    // Convert parameters to JSON (May work on this feature later)
+	String json = convertToJson(name,email,text);
+
+    // Add input to ArrayList 
+    if(!text.equals("")){
+      comments.add(text + "\n"); 
+    }
+
+    // Send the HTML as the response
     response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello Jamie!</h1>");
+    for(int i = 0; i < comments.size(); i++){
+      response.getWriter().println(comments.get(i));
+    }
+
+    // Send user to new page once comment is submitted
+    response.sendRedirect("/index.html");
```

Comment:
This line will always redirect the page to the main index page after a comment is left, meaning that when someone leaves a comment on "My Hidden Talent," for example, they're taken off of that page and to the main page when they hit submit.

This is fine for now, but if you want to work further on the comment functionality after you finish all of the week's goals, fixing this should probably be one of your top priorities.